alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -703,429 +507,45 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
-                    "and queryIndex [${queryIndices.joinToString()}] for ${docs.size} document(s)",
-                e
-            )
-        }
-
-        if (response.status() !== RestStatus.OK) {
-            throw IOException(
-                "Monitor ${monitor.id}: Failed to search percolate index: ${queryIndices.joinToString()}. " +
-                    "Response status is ${response.status()}"
-            )
-        }
-        logger.debug("Monitor ${monitor.id} PERF_DEBUG: Percolate query time taken millis = ${response.took}")
-        percolateQueriesTimeTakenStat += response.took.millis
-        return response.hits
-    }
-    /** we cannot use terms query because `index` field's mapping is of type TEXT and not keyword. Refer doc-level-queries.json*/
-    private fun buildShouldClausesOverPerIndexMatchQueries(indices: List<String>): BoolQueryBuilder {
-        val boolQueryBuilder = QueryBuilders.boolQuery()
-        indices.forEach { boolQueryBuilder.should(QueryBuilders.matchQuery("index", it)) }
-        return boolQueryBuilder
-    }
+    ): Map<String, MutableSet<ShardId>> {
 
-    /** Transform field names and index names in all the search hits to format required to run percolate search against them.
-     * Hits are transformed using method transformDocumentFieldNames() */
-    private fun transformSearchHitsAndReconstructDocs(
-        hits: SearchHits,
-        index: String,
-        concreteIndex: String,
-        monitorId: String,
-        conflictingFields: List<String>,
-    ): List<Pair<String, TransformedDocDto>> {
-        return hits.mapNotNull(fun(hit: SearchHit): Pair<String, TransformedDocDto>? {
-            try {
-                val sourceMap = if (hit.hasSource()) {
-                    hit.sourceAsMap
-                } else {
-                    constructSourceMapFromFieldsInHit(hit)
-                }
-                transformDocumentFieldNames(
-                    sourceMap,
-                    conflictingFields,
-                    "_${index}_$monitorId",
-                    "_${concreteIndex}_$monitorId",
-                    ""
-                )
-                var xContentBuilder = XContentFactory.jsonBuilder().map(sourceMap)
-                val sourceRef = BytesReference.bytes(xContentBuilder)
-                docsSizeOfBatchInBytes += sourceRef.ramBytesUsed()
-                totalDocsSizeInBytesStat += sourceRef.ramBytesUsed()
-                return Pair(hit.id, TransformedDocDto(index, concreteIndex, hit.id, sourceRef))
-            } catch (e: Exception) {
-                logger.error("Monitor $monitorId: Failed to transform payload $hit for percolate query", e)
-                // skip any document which we fail to transform because we anyway won't be able to run percolate queries on them.
-                return null
-            }
-        })
-    }
+        val totalShards = shards.size
+        val numFanOutNodes = allNodes.size.coerceAtMost((totalShards + 1) / 2)
```

Comment:
I might have missed it earlier in the code, but what's the need to limit the number of fan out nodes to half the shard count?